### PR TITLE
fix(rollup): use workspace rootDir while compiling TS files

### LINKED ISSUE #11289: 'rootDir' is expected to contain all source files. Bundle failed: test-icons
## Current Behavior

Now when I run `build` command in our repo with multiple publishable libs, I get error like this:

```bash
test-icons:build-lib

Bundling test-icons...
Error during bundle: Error: /Users/stupic/Documents/demos/build-lib-bug-repro/libs/test-icons/src/index.ts(2,15): semantic error TS6059: File '/Users/stupic/Documents/demos/build-lib-bug-repro/libs/test-core/src/index.ts' is not under 'rootDir' '/Users/stupic/Documents/demos/build-lib-bug-repro/libs/test-icons/src'. 'rootDir' is expected to contain all source files.
Bundle failed: test-icons

 ——————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  NX   Ran target build-lib for project test-icons (3s)

    ✖    1/1 failed
    ✔    0/1 succeeded [0 read from cache]
```

## Expected Behavior

Expect to build successfully or with a better error message like: `test-core` has no `build-lib` target.

## Steps to Reproduce

1. clone the repro repo:
```
git clone git@github.com:SunStupic/build-lib-bug-repro.git
```
2. `pnpm i`
3. `nx run  test-icons:build-lib`

### Failure Logs

```bash
test-icons:build-lib

Bundling test-icons...
Error during bundle: Error: /Users/stupic/Documents/demos/build-lib-bug-repro/libs/test-icons/src/index.ts(2,15): semantic error TS6059: File '/Users/stupic/Documents/demos/build-lib-bug-repro/libs/test-core/src/index.ts' is not under 'rootDir' '/Users/stupic/Documents/demos/build-lib-bug-repro/libs/test-icons/src'. 'rootDir' is expected to contain all source files.
Bundle failed: test-icons

 ——————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  NX   Ran target build-lib for project test-icons (3s)

    ✖    1/1 failed
    ✔    0/1 succeeded [0 read from cache]
```

### Environment

```
 nx report

 >  NX   Running global Nx CLI with PNPM may have issues.

   Prefer to use "pnpm" (https://pnpm.io/cli/exec) to execute commands in this workspace.
    TIP  create a shortcut such as: alias pnx="pnpm nx --"



 >  NX   Report complete - copy this into the issue template

   Node : 16.15.1
   OS   : darwin x64
   pnpm : 7.5.0

   nx : 14.4.3
   @nrwl/angular : Not Found
   @nrwl/cypress : 14.4.3
   @nrwl/detox : Not Found
   @nrwl/devkit : Not Found
   @nrwl/eslint-plugin-nx : 14.4.3
   @nrwl/express : Not Found
   @nrwl/jest : 14.4.3
   @nrwl/js : Not Found
   @nrwl/linter : 14.4.3
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : 14.4.3
   @nrwl/react-native : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : Not Found
   @nrwl/web : 14.4.3
   @nrwl/workspace : 14.4.3
   typescript : 4.7.4
   ---------------------------------------
   Community plugins:

```

## Workaround

Now the workaround is simple, we need to add `build-lib` for all the publishable packages or we change back to use `build`.

### PR DESCRIPTION
**Pull request by kopach**
Fixes #11289

Tested this approach on real project using [pnpm's patch mechanism](https://pnpm.io/cli/patch)

### CODE REVIEW

File: packages/rollup/src/executors/rollup/rollup.impl.ts [hunk 1]
```diff
@@ -306,7 +306,7 @@ function createTsCompilerOptions(
 ) {
   const compilerOptionPaths = computeCompilerOptionsPaths(config, dependencies);
   const compilerOptions = {
-    rootDir: options.projectRoot,
+    rootDir: config.options.rootDir,
```
Comment (kopach) [inline packages/rollup/src/executors/rollup/rollup.impl.ts:309]: `options.projectRoot` contains path to root of particular package. `config.options.rootDir` contain path to workspace root.

**Comment by kopach** [comment]
hi @ndcunningham, could you, please, advice how can I handle the CI failing issue? I'ts not clear for me what exactly failing and why

**Review by mandarini — APPROVED** [review]

**Comment by mandarini** [comment]
Hello @kopach ! I looked into this thoroughly and that PR is not achieving the desired behaviour. What is described in the issue, is that if the `build` target name is something else, then Nx does not identify the library as buildable, so this is the issue that should be addressed.

I will be closing this PR, but thank you very much for contributing! Let me know if you have any concerns!

**Comment by kopach** [comment]
Ok, sure @mandarini . No problem. It's just something which solves my issue (which has exact error output as described). I'm patching nx locally and it does the trick for me

**Pull request closed without merging**